From 11856a62d45596d526987f333525cd03f05aec03 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Fri, 15 Aug 2014 10:19:16 +0800
Subject: [PATCH 4/4] arm: omap: wkup_m3: invoke release_firmware() in
 wkup_m3_firmware_cb()

We should invoke the release_firmware() to free the memory allocated
for the firmware loading in the callback functions. The following is
the call trace reported by kmemleak:
unreferenced object 0xed8bccc0 (size 64):
  comm "kworker/0:1", pid 20, jiffies 4294937580 (age 190.450s)
  hex dump (first 32 bytes):
    2c 2a 00 00 54 60 76 c0 00 00 00 00 00 00 00 00  ,*..T`v.........
    00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
  backtrace:
    [<c0158f5c>] create_object+0x130/0x294
    [<c06ebe78>] kmemleak_alloc+0x48/0x78
    [<c014b160>] kmem_cache_alloc_trace+0x17c/0x3d4
    [<c04044b8>] _request_firmware+0x50/0x85c
    [<c0404de4>] request_firmware_work_func+0x38/0x68
    [<c0066fbc>] process_one_work+0x1c8/0x6ec
    [<c00682bc>] worker_thread+0x144/0x3f8
    [<c006f9c8>] kthread+0xb4/0xb8
    [<c000e4c8>] ret_from_fork+0x14/0x20
    [<ffffffff>] 0xffffffff

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/mach-omap2/wkup_m3.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/wkup_m3.c b/arch/arm/mach-omap2/wkup_m3.c
index c35327f..5e693b9 100644
--- a/arch/arm/mach-omap2/wkup_m3.c
+++ b/arch/arm/mach-omap2/wkup_m3.c
@@ -346,6 +346,7 @@ static void wkup_m3_firmware_cb(const struct firmware *fw, void *context)
 		wkup_m3->is_valid = true;
 	}
 
+	release_firmware(fw);
 	return;
 }
 
-- 
1.7.5.4

